src/main/java/io/pravega/connectors/flink/Pravega.java

```diff
@@ -0,0 +1,332 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.flink;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.stream.Stream;
+import io.pravega.client.stream.StreamCut;
+import io.pravega.connectors.flink.util.StreamWithBoundaries;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.descriptors.ConnectorDescriptor;
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.flink.table.descriptors.ConnectorDescriptorValidator.CONNECTOR_VERSION;
+
+/**
+ * Pravega connector descriptor.
+ */
+
+public class Pravega extends ConnectorDescriptor {
+
+    public static final String CONNECTOR_TYPE_VALUE_PRAVEGA = "pravega";
+
+    // currently the value is fixed but as we evolve we may have multiple versions to support
+    public static final int CONNECTOR_VERSION_VALUE = 1;
+
+    public static final String CONNECTOR_METRICS = "connector.metrics";
+
+    // Connection Configurations
+    public static final String CONNECTOR_CONNECTION_CONFIG = "connector.connection-config";
```

Comment:
Wondering if these names could be shortened a little bit. For instance, all of them are about the table connector, so maybe the `CONNECTOR` prefix in all these constants is not providing much information. Another example: we only use Reader Groups in the Reader, so maybe defining `READER_GROUP` instead of `CONNECTOR_READER_READER_GROUP` could be enough.

Reply:
These names are defined to match the corresponding Pravega YAML configuration element names. Similar naming convention is followed for other connectors.